src/Transition_Systems-Weak_Transition_Systems.thy

```diff
@@ -36,6 +36,20 @@ notation weak.constant_bisimilarity (\<open>[\<approx>]\<close>)
 notation weak.simulation_up_to (\<open>weak'_sim\<^bsub>_\<^esub>\<close>)
 notation weak.bisimulation_up_to (\<open>weak'_bisim\<^bsub>_\<^esub>\<close>)
 
+lemma mutual_silent_weak_transitions_up_to_bisimilarity:
+  assumes "p \<Rightarrow>\<lparr>\<tau>\<rparr> p'" and "p' \<approx> q" and "q \<Rightarrow>\<lparr>\<tau>\<rparr> q'" and "q' \<approx> p"
+  shows "p \<approx> q"
+using assms
+proof (coinduction arbitrary: p q p' q' rule: weak.symmetric_up_to_rule [where \<F> = "[\<approx>]"])
+  case (simulation \<alpha> s)
```

Comment:
Could you extend this to `case (simulation α s p q p' q')`? It’s not strictly necessary, but corresponds to what I’ve done throughout the transition system framework and the Þ-calculus formalization.

Reply:
I think `s'` should rather be `t`, according to the conventions used in similar proofs.

Reply:
The chained fact `q ⇒⦇τ⦈ q'` is not needed and should thus be dropped.

Reply:
I think it’s better to conclude `s ≈ t` instead of `t ≈ s` (`s' ≈ s`), as this better corresponds to the normal statement of `(≈)` being a simulation relation (the bisimilarity used as a chained fact has sides swapped, but this is because this way it looks better in the lemma statement).

Reply:
When switching from `t ≈ s` to `s ≈ t`, as described in the previous comment, the proof of this step doesn’t work anymore, but the following, perhaps even nicer, proof does:
```isabelle
using weak.bisimilarity_is_simulation
by (blast dest: weak.bisimilarity_symmetry_rule)
```

Reply:
Oops, I just found out that all my comments to this line except the first one are actually misplaced: they refer to the next line. Sorry.

Reply:
> Could you extend this to `case (simulation α s p q p' q')`? It’s not strictly necessary, but corresponds to what I’ve done throughout the transition system framework and the Þ-calculus formalization.

Fixed in 4d1256b61eec4c2d2e641cce8a67c32ab97a30e7, please verify.

Reply:
> I think `s'` should rather be `t`, according to the conventions used in similar proofs.

Fixed in 0e6786045df266cc34e2717369b474ad7e6a79e9, please verify.

Reply:
> The chained fact `q ⇒⦇τ⦈ q'` is not needed and should thus be dropped.

Fixed in 3c0f6d6077c9fb9f2a89bd2e175c96887fd44df3, please verify.

Reply:
> I think it’s better to conclude `s ≈ t` instead of `t ≈ s` (`s' ≈ s`), as this better corresponds to the normal statement of `(≈)` being a simulation relation (the bisimilarity used as a chained fact has sides swapped, but this is because this way it looks better in the lemma statement).

Fixed in 048cd571e1f663f759ccdd46df32e32b1e2bf7f5, please verify.

Reply:
> When switching from `t ≈ s` to `s ≈ t`, as described in the previous comment, the proof of this step doesn’t work anymore, but the following, perhaps even nicer, proof does:
> 
> ```isabelle-theory
> using weak.bisimilarity_is_simulation
> by (blast dest: weak.bisimilarity_symmetry_rule)
> ```

Also fixed in 048cd571e1f663f759ccdd46df32e32b1e2bf7f5, please verify.

Reply:
All this looks good to me.